src/tasks/task-se3-equality.cpp

```diff
@@ -153,37 +160,45 @@ namespace tsid
       m_robot.frameVelocity(data, m_frame_id, v_frame);
       m_robot.frameClassicAcceleration(data, m_frame_id, m_drift);
 
-      // Transformation from local to world
-      m_wMl.rotation(oMi.rotation());
+      // @todo Since Jacobian computation is cheaper in world frame
+      // we could do all computations in world frame
+      m_robot.frameJacobianLocal(data, m_frame_id, m_J);
 
       errorInSE3(oMi, m_M_ref, m_p_error);          // pos err in local frame
-      m_v_error = v_frame - m_wMl.actInv(m_v_ref);  // vel err in local frame
-
       m_p_error_vec = m_p_error.toVector();
-      m_v_error_vec = m_v_error.toVector();
       SE3ToVector(m_M_ref, m_p_ref);
-      m_v_ref_vec = m_v_ref.toVector();
       SE3ToVector(oMi, m_p);
-      m_v = v_frame.toVector();
 
-#ifndef NDEBUG
-//      PRINT_VECTOR(v_frame.toVector());
-//      PRINT_VECTOR(m_v_ref.toVector());
-#endif
+      // Transformation from local to world
+      m_wMl.rotation(oMi.rotation());
 
-      // desired acc in local frame
-      m_a_des = - m_Kp.cwiseProduct(m_p_error.toVector())
-                - m_Kd.cwiseProduct(m_v_error.toVector())
-                + m_wMl.actInv(m_a_ref).toVector();
+      if (m_local_frame) {
+        m_v_error = v_frame - m_wMl.actInv(m_v_ref);  // vel err in local frame
 
-      // @todo Since Jacobian computation is cheaper in world frame
-      // we could do all computations in world frame
-      m_robot.frameJacobianLocal(data, m_frame_id, m_J);
+        // desired acc in local frame
+        m_a_des = - m_Kp.cwiseProduct(m_p_error.toVector())
+                  - m_Kd.cwiseProduct(m_v_error.toVector())
+                  + m_wMl.actInv(m_a_ref).toVector();
+      } else {
+        m_v_error = m_wMl.act(v_frame) - m_v_ref;  // vel err in local oriented frame.
+
+        m_drift = m_wMl.act(m_drift);
+
+        // desired acc in local oriented frame
+        m_a_des = - m_Kp.cwiseProduct(m_p_error.toVector())
```

Comment:
Shouldn't you rotate `m_p_error` from local to global before using it here?

Reply:
Unfortunately my memory is not very good, so I don't recall this chat. Anyway, the comment in line 177, where `m_p_error` is computed, says "pos err in local frame", so I would trust it. Moreover, at the moment you're using `m_p_error` in the same way regardless of which frame you're doing computations in (see lines 179 and 188). This is clearly wrong, and one of them must be rotated. I think that this is the one that should be rotated, whereas line 179 is fine.